[Low Kaiwei] iP

lowkaiwei98 opened the pull request:
# Tsundere

> Soon is not as good as now. – Seth Godin ([source](https://dansilvestre.com/productivity-quotes/))

Tsundere frees your mind of having to remember things you need to do while scolding you for forgeting. It's
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/lowkaiwei98/ip/releases/tag/A-jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smiley:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Hot waifu anime girl (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the constructor for `Tsundere`:

```java
public Tsundere(String filePath) {
        ui = new Ui();
        storage = new Storage(filePath);
        try {
            tasks = new TaskList(storage.load());
        } catch (TsundereException e) {
            ui.showLoadingError();
            tasks = new TaskList();
        }
    }
```

Commit:
Add Gradle support

Commit:
Level-1 Greet, Echo, Exit

Commit:
Update for Level 2, add list

Commit:
Added Task for level 3

Commit:
Added Todo, dealine and event for level 4

Commit:
Fixed ui test bat

Commit:
Added exception for level 5

Commit:
Added the ability to delete tasks for Level 6

Commit:
Changed part of the code to enum

Commit:
Changed input and expected for textUiTest

Commit:
Add the save functionability

Commit:
Add the ability to read time in the format yyyy-mm-dd

Commit:
Merge branch 'branch-Level-8'

Commit:
Change code to OOP paradigm

Commit:
Add packages according to different classes

Commit:
Add Junit test

Commit:
Fix various bugs regrading saving file and release jar file

Commit:
Add Java Docs

Commit:
Tweak code to comply with the coding standard

Commit:
Add find ability which give users a way to find a task by searching for a keyword

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Task/TaskList.java

Commit:
Merge branch 'branch-Level-9'

Commit:
Fix merge issues

Commit:
Update README.md

Commit:
Update GFMD in the PR description

Commit:
Merge branch 'master' of https://github.com/lowkaiwei98/ip

# Conflicts:
#	README.md

Commit:
Fix GFMD issues

Commit:
Fix GFMD issues

Commit:
Fix packing naming

Commit:
Merge branch 'add-gradle-support'

Commit:
Fix coding style violations while add checkStyle

Commit:
Add a GUI to Tsundere. Use the JavaFX technology to implement the GUI.

Commit:
User Java varargs in finding tasks for a better search result

nicollegann commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,93 @@
+     * @param t TaskList for managing and adding tasks
+     * @param u UI for displaying text
+     * @param s Storage for saving to file
+     * @throws TsundereException for incorrect input format
```
Should the first sentence be 'Executes' and 'stores'? Also, I think there should be a punctuation behind each parameter description. I noticed this in several other javadocs too

nicollegann commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,93 @@
+        EVENT
+    }
+
+    protected Type ty;
```
Perhaps naming the variable as `type` instead of `ty` may sound more intuitive?

lowkaiwei98 replied:
You're right! i changed it to type.

nicollegann commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    /**
+     * string format for the save file
```
Will it be clearer to write 'Formats string for the save file' or 'Returns string format for the save file'? I think this is repeated in several other `toString()` methods too

lowkaiwei98 replied:
Yes 'Returns string format for the save file' sounds better. I changed it, thanks.

nicollegann commented on src/main/java/tsundere/Parser.java:
```diff
@@ -0,0 +1,86 @@
+     */
+    public static Command parse(String s) throws TsundereException, NumberFormatException {
+
+        String us = s.toUpperCase();
```
Might want to use a clearer variable name

nicollegann commented on src/main/java/tsundere/Storage.java:
```diff
@@ -0,0 +1,114 @@
+            return emptyTasks;
+        }
+        Scanner c = new Scanner(returnStr);
+        int countLst = 0;
```
Not sure if Lst in `countLst` represents List or is that an abbreviation for another word 😅

nicollegann commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,93 @@
+ * Add a new task into tasklist and storage saves it
+ */
+public class AddCommand extends Command {
+    enum Type {
+        TODO,
+        DEADLINE,
+        EVENT
+    }
```
I like how you used enums here!

lowkaiwei98 replied:
Thanks

nicollegann commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,37 @@
+        s.saveFile(t.tasksToString());
+    }
+
+    public boolean isExit() {
+        return false;
+    }
```
I think you might have missed out the Javadoc comments for this method

nicollegann reviewed (commented):
I find your project to be nicely structured and easy to read! Other than some minor improvements, I think your code abides by most of the code quality guidelines. Overall, a solid piece of work! 😄👍

Commit:
Fix javadoc and code quality issues

Commit:
Fixed exitcommand, now actually closes the problem after 3 secs

Commit:
Update with assert feature

With assertions, runtime can verify assumptions about the program state

Add assert at:
- DeleteCommand, for count of TaskLists

Commit:
Refactor code to improve code quality.

Critically examines the code and refactor to improve the code quality where necessary.

Major changes at:
-AddCommand
-Parser

Commit:
Merge pull request #1 from lowkaiwei98/branch-A-Assertions

Update with assert feature

Commit:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Merge pull request #2 from lowkaiwei98/branch-A-CodeQuality

Refactor code to improve code quality.

Commit:
Merge branch 'master' of https://github.com/lowkaiwei98/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/command/DeleteCommand.java

Commit:
Add an extension and provide a way to get reminders about tasks

Commit:
Set up a product website

Commit:
Set theme jekyll-theme-cayman
